templates/SilverStripe/CMS/Forms/SiteTreeURLSegmentField.ss

```diff
@@ -1,20 +1,31 @@
+<% if not $IsReadonly %>
 <div class="preview-holder">
-	<a class="URL-link" href="$URL" target="_blank">
-		$URL
-	</a>
-	<% if not $IsReadonly %>
-		<button role="button" type="button" class="btn btn-outline-secondary btn-sm edit">
-			<%t SilverStripe\CMS\Forms\SiteTreeURLSegmentField.Edit 'Edit' %>
-		</button>
-	<% end_if %>
+    <a class="URL-link" href="$URL" target="_blank">
+        $URL
+    </a>
+    <button role="button" type="button" class="btn btn-outline-secondary btn-sm edit">
+        <%t SilverStripe\CMS\Forms\SiteTreeURLSegmentField.Edit 'Edit' %>
+    </button>
 </div>
+
 <div class="edit-holder">
-	<input $AttributesHTML />
-	<button role="button" data-icon="accept" type="button" class="btn btn-primary btn-sm update">
-		<%t SilverStripe\CMS\Forms\SiteTreeURLSegmentField.OK 'OK' %>
-	</button>
-	<button role="button" data-icon="cancel" type="button" class="btn btn-outline-secondary btn-sm cancel">
-		<%t SilverStripe\CMS\Forms\SiteTreeURLSegmentField.Cancel 'Cancel' %>
-	</button>
-	<% if $HelpText %><p class="form__field-description">$HelpText</p><% end_if %>
+    <div class="input-group">
+        <input $AttributesHTML />
+        <div class="input-group-append">
+            <button role="button" data-icon="accept" type="button" class="btn btn-primary update">
+                <%t SilverStripe\CMS\Forms\SiteTreeURLSegmentField.OK 'OK' %>
+            </button>
+        </div>
+        <div class="input-group-append">
+            <button role="button" data-icon="cancel" type="button" class="btn btn-outline-secondary btn-sm input-group-append cancel">
+                <%t SilverStripe\CMS\Forms\SiteTreeURLSegmentField.Cancel 'Cancel' %>
+            </button>
+        </div>
+    </div>
+    <% if $HelpText %><p class="form__field-description">$HelpText</p><% end_if %>
 </div>
+<% else %>
+<a class="readonly-url" href="$URL" target="_blank">
```

Comment:
When I click this link I get taken to the live URL. The old form field would take me to the Draft URL.

Reply:
The current field will post back a value when submitted in read only mode. This new one will not.

It's doesn't look like it's breaking anything on the CMS Form, but it would be safer to replicate the original behavior.

Reply:
I've reverted to the old behaviour of just hiding the Edit action, thus the field is still present.